db/src/tx.rs

```diff
@@ -340,25 +341,20 @@ impl<'conn, 'a> Tx<'conn, 'a> {
     }
 }
 
-use std::borrow::Cow;
-
-/// Transact the given `entities` against the given SQLite `conn`, using the metadata in
-/// `self.DB`.
+/// Transact the given `entities` against the given SQLite `conn`, using the given metadata.
 ///
 /// This approach is explained in https://github.com/mozilla/mentat/wiki/Transacting.
 // TODO: move this to the transactor layer.
-pub fn transact<'conn, 'a, I>(conn: &'conn rusqlite::Connection, partition_map: &'a PartitionMap, schema: &'a Schema, entities: I) -> Result<(TxReport, PartitionMap, Option<Schema>)> where I: IntoIterator<Item=Entity> {
+pub fn transact<'conn, 'a, I>(conn: &'conn rusqlite::Connection, mut partition_map: PartitionMap, schema: &'a Schema, entities: I) -> Result<(TxReport, PartitionMap, Option<Schema>)> where I: IntoIterator<Item=Entity> {
```

Comment:
Why does `partition_map` need to be `mut`? IIRC the only reason for that here is if you assign to `partition_map` in the body of the function, and you don't, right?